packages/falter-berlin-service-registrar/files/register-services.sh

```diff
@@ -0,0 +1,241 @@
+
+    # generate service entry in olsrd-config
+    uci_add_list olsrd @LoadPlugin["$nsplugin"] hosts "$host_announcment"
+    uci_add_list olsrd @LoadPlugin["$nsplugin"] service "$service_announcement"
+}
+
+apply_configs() {
+    uci commit uhttpd
+    uci_commit olsrd
+    /etc/init.d/uhttpd restart
+    /etc/init.d/olsrd restart
+    /etc/init.d/dnsmasq restart
+}
+
+handle_service() {
+    local uci_section="$1"
+    #local custom="$2"
+
+    # 1. Name of a variable to store the retrieved value in
+    # 2. ID of the section to read the value from
+    # 3. Name of the option to read the value from
+    # 4. Default (optional), value to return instead if option is unset
+    # see https://openwrt.org/docs/guide-developer/config-scripting
+
+    config_get fqdn "$uci_section" fqdn
+    config_get description "$uci_section" description
+    config_get protocol "$uci_section" protocol "tcp"
+    config_get port "$uci_section" port "80"
+    config_get ip_addr "$uci_section" ip_addr
+    config_get disabled "$uci_section" disabled "0"
+
+    if [ "${disabled:?}" = 0 ]; then
+
+        if [ -z "$fqdn" ] || [ -z "$description" ] || [ -z "$ip_addr" ]; then
+            log "service configuration failed! UCI-section $uci_section misses at least one of mandatory fqdn, description or ip_addr options"
+            # return 0 to allow configuration of other services
+            return 0
+        fi
+
+        log "registering service with
+    FQDN.................: $fqdn
+    Name/Description.....: $description
+    Protocol.............: $protocol
+    Port.................: $port
+    IP Address...........: $ip_addr"
+
+        register_service_at_olsrd "$ip_addr" "$fqdn" "$port" "$protocol" "$description"
+
+    fi
+}
+
+handle_website() {
+    local uci_section="$1"
+    #local custom="$2"
+
+    # see https://openwrt.org/docs/guide-developer/config-scripting
+
+    config_get fqdn "$uci_section" fqdn
+    config_get description "$uci_section" description
+    config_get protocol "$uci_section" protocol "tcp"
+    config_get port "$uci_section" port "80"
+    config_get web_root "$uci_section" web_root
+    config_get disabled "$uci_section" disabled "0"
+
+    if [ "$disabled" = 0 ]; then
+
+        if [ -z "$fqdn" ] || [ -z "$description" ] || [ -z "$web_root" ]; then
+            log "service configuration failed! UCI-section $uci_section misses at least one of mandatory fqdn, description or web_root options"
+            # return 0 to allow configuration of other services
+            return 0
+        fi
+
+        log "registering website with
+    FQDN.................: $fqdn
+    Name/Description.....: $description
+    Protocol.............: $protocol
+    Port.................: $port
+    Web root.............: $web_root"
+
+        create_website_template "$web_root"
+
+        register_website_at_uhttpd "$uci_section" "$fqdn" "$port" "$web_root"
+
+        website_addr=$(uci_get network dhcp ipaddr) # first IP-addr of dhcp-subnet
```

Comment:
In general no "-" in english, and abbreviations in capital letters:
IP address
DHCP subnet

I probably missed some too... ;-)

Reply:
Adjusted in new commit version.